### PR TITLE
[SMTChecker] Clear state knowledge after external function calls

### LINKED ISSUE #5190: SMTChecker: reset state variables after external call to unknown function

### PR DESCRIPTION
**Pull request by leonardoalt**
Fixes #5190

### CODE REVIEW

**Review by chriseth — CHANGES_REQUESTED** [review]
Please add a test that shows that mappings (also local variables referring to state variables) are properly reset.

**Comment by leonardoalt** [comment]
@chriseth Added tests

File: Changelog.md [hunk 1]
```diff
@@ -5,6 +5,7 @@ Language Features:
 
 Compiler Features:
  * Control Flow Graph: Warn about unreachable code.
+ * SMTChecker: Support external function calls in an abstract way.
```
Comment (chriseth) [inline Changelog.md]: Could you make this more specific (;-))?

File: libsolidity/formal/SMTChecker.cpp [hunk 2]
```diff
@@ -412,6 +412,10 @@ void SMTChecker::endVisit(FunctionCall const& _funCall)
 	case FunctionType::Kind::Internal:
 		inlineFunctionCall(_funCall);
 		break;
+	case FunctionType::Kind::External:
+		resetStateVariables();
+		resetReferences();
```
Comment (chriseth) [inline libsolidity/formal/SMTChecker.cpp]: Should be `resetStorageReferences` (even if only storage references have been implemented so far).

**Review by chriseth — CHANGES_REQUESTED** [review]

File: Changelog.md (hunk 1 above)
Comment (leonardoalt) [reply Changelog.md]: Done

**Comment by chriseth** [comment]
Needs rebase.

**Review by chriseth — APPROVED** [review]

**Pull request merged**